File: benchmarks/runtime/rails/db/migrate/20240101000001_create_posts.rb
class CreatePosts < ActiveRecord::Migration[7.2]
  def change
    create_table :posts do |t|
      t.text    :title,     null: false
      t.text    :body,      null: false
      t.boolean :published, null: false, default: false
      t.text    :author,    null: false
      t.timestamps null: false
    end

    create_table :api_tokens do |t|
      t.text      :token,     null: false
      t.text      :principal, null: false
      t.timestamp :created_at, null: false, default: -> { "NOW()" }
    end
    add_index :api_tokens, :token, unique: true
  end
end

File: benchmarks/runtime/phoenix/priv/repo/migrations/20240101000001_create_posts.exs
defmodule Benchmark.Repo.Migrations.CreatePosts do
  use Ecto.Migration

  def change do
    create table(:posts) do
      add :title,     :text,    null: false
      add :body,      :text,    null: false
      add :published, :boolean, null: false, default: false
      add :author,    :text,    null: false
      timestamps(inserted_at: :created_at, type: :utc_datetime)
    end

    create table(:api_tokens) do
      add :token,     :text, null: false
      add :principal, :text, null: false
      timestamps(inserted_at: :created_at, updated_at: false, type: :utc_datetime)
    end

    create unique_index(:api_tokens, [:token])
  end
end
